### PR TITLE
CBL-5448: Attachments flag dropped when applying delta to incoming rev

### PR DESCRIPTION
**Pull request by callumbirks**
`hasBlobReferences` is copied in from master.
You can show the fix works locally, if you comment out the 3 lines added in `IncomingRev.cc`, the test fails

### CODE REVIEW

File: Replicator/IncomingRev.cc [hunk 1]
```diff
@@ -359,6 +359,10 @@ namespace litecore { namespace repl {
             }
         }
 
+        if (!(_rev->flags & kRevHasAttachments) && _db->hasBlobReferences(root)) {
+            _rev->flags |= kRevHasAttachments;
+        }
```
Comment (pasin) [inline Replicator/IncomingRev.cc:364]: As commented in the slack, I think we should reuse the logic in line 330 - 343, as we don't need to find blob for the case that delta sync is not used AND no blobs changes.

The change could be : 

```
if (_mayContainBlobChanges || <didApplyDelta>) {
            _db->findBlobReferences(root, true, [=](FLDeepIterator i, Dict blob, const C4BlobKey &key) {
                _rev->flags |= kRevHasAttachments;
                if (_mayContainBlobChanges) {
                   _pendingBlobs.push_back({_rev->docID,
                                         alloc_slice(FLDeepIterator_GetPathString(i)),
                                         key,
                                         blob["length"_sl].asUnsigned(),
                                         C4Blob::isLikelyCompressible(blob)});
                   _blob = _pendingBlobs.begin();
                }
            });
}
```

**Review by pasin — CHANGES_REQUESTED** [review]

File: Replicator/IncomingRev.cc [hunk 2]
```diff
@@ -218,6 +221,7 @@ namespace litecore { namespace repl {
         // First create a Fleece document:
         Doc fleeceDoc;
         C4Error err = {};
+        bool didApplyDelta;
```
Comment (jianminzhao) [inline Replicator/IncomingRev.cc:224]: initializae

Comment (callumbirks) [reply Replicator/IncomingRev.cc:224]: Booleans are already initialized (so are all integral types).
Just structs, classes, etc are uninitialized

Comment (jianminzhao) [reply Replicator/IncomingRev.cc:224]: I doubt it. At least GPT does not think so.

Comment (callumbirks) [reply Replicator/IncomingRev.cc:224]: Oh I see, Xcode doesn't like it regardless of that.

**Review by jianminzhao — APPROVED** [review]

**Review by pasin — APPROVED** [review]

**Review by pasin — APPROVED** [review]

**Pull request merged**